lib/kafka/offset_manager.rb

```diff
@@ -111,17 +116,51 @@ def seconds_since_last_commit
       Time.now - @last_commit
     end
 
+    def seconds_since_last_recommit
+      Time.now - @last_recommit
+    end
+
     def committed_offset_for(topic, partition)
       @committed_offsets ||= @group.fetch_offsets
       @committed_offsets.offset_for(topic, partition)
     end
 
+    def offsets_to_commit(recommit = false)
+      if recommit && !@committed_offsets.nil?
```

Comment:
If `@committed_offsets` is set to `nil` in `commit_offsets` won't this only recommit the first time `recommit = true` is passed? Also how are `@committed_offsets` and `@resolved_offsets` different?